twine/repository.py

```diff
@@ -116,14 +114,14 @@ def set_client_certificate(self, clientcert: Optional[str]) -> None:
         if clientcert:
             self.session.cert = clientcert
 
-    def register(self, package):
-        data = package.metadata_dictionary()
+    def register(self, pkg):
```

Comment:
Beyond what I already mentioned, this is considered a public interface and changing the naming of an argument is a backwards-incompatible change.